src/jailer/src/env.rs

```diff
@@ -481,4 +498,44 @@ mod tests {
             libc::close(fd2);
         }
     }
+
+    #[test]
+    fn test_setup_jailed_folders() {
+        let arg_parser = build_arg_parser();
+        let mut args = arg_parser.arguments().clone();
+        args.parse(&make_args(&ArgVals::new())).unwrap();
+        let env = Env::new(&args, 0, 0).unwrap();
+
+        // Error case: non UTF-8 paths.
+        let bad_string: &[u8] = &[0, 102, 111, 111]; // A leading nul followed by 'f', 'o', 'o'
+        let bad_osstr = unsafe { &*(bad_string as *const [u8] as *const OsStr) };
+        assert_eq!(
+            format!("{}", env.setup_jailed_folders(vec![bad_osstr]).err().unwrap()),
+            "Failed to decode string from byte array: data provided contains an interior nul byte at byte pos 0"
+        );
+
+        // Error case: inaccessible path - can't be triggered with unit tests running as root.
+        // assert_eq!(
+        //     format!("{}", env.setup_jailed_folders(vec!["/foo/bar"]).err().unwrap()),
+        //     "Failed to create directory /foo/bar: Permission denied (os error 13)"
+        // );
+
+        // Success case.
+        let foo_dir = TempDir::new().unwrap();
+        let foo_path = foo_dir.as_path();
+        foo_dir.remove().unwrap();
+        assert!(env.setup_jailed_folders(vec![foo_path]).is_ok());
+
+        let metadata = fs::metadata(foo_path).unwrap();
+        // The mode bits will also have the directory bit set.
```

Comment:
Is this comment referring by `directory bits` to the `setuid`, `setgid` and `sticky bit` attributes?
As far as I understand `0x4000` means that `setgid` and `sticky bit` is on. Why would these two attributes be required?

Reply:
Sorry, just `sergid` actually.

Reply:
Nope, it's `S_IFDIR`, the bit asserted by `S_ISDIR` (see the manpage for `stat`, or [here it is in `musl`](https://git.musl-libc.org/cgit/musl/tree/include/sys/stat.h#n31)). I'll reword the comment to make this more clear.

Reply:
It's actually better to use `libc::S_IFDIR` directly instead of hardcoding some hex value, so I did that instead, PTAL.

Reply:
Makes sense. Seems ok.